src/package-hoister.js

```diff
@@ -242,7 +242,7 @@ export default class PackageHoister {
           continue;
         }
 
-        const isMarkedAsOptional = !depinfo.pkg._reference || this.ignoreOptional;
+        const isMarkedAsOptional = depinfo.pkg._reference && depinfo.pkg._reference.optional && this.ignoreOptional;
```

Comment:
sorry I deleted the previous comment because I realized the logic isn't quite the same. The previous one is saying mark as optional if `depinfo.pkg._reference` doesn't exist (or if ignoreOptional flag). That seemed a little weird to be but if that's still the case, we should probably restore it to the way you had it.

Reply:
I know they are not equivalent but no tests failed and I feel like it makes more sense with the new version. What do you think?

Reply:
Works for me. LET'S LIVE ON THE EDGE. 🎉